fig/service.py

```diff
@@ -408,14 +422,15 @@ def build(self, no_cache=False):
                     image_id = match.group(1)
 
         if image_id is None:
-            raise BuildError(self)
+            raise BuildError(self, event if all_events else 'Unknown')
```

Comment:
Using the `event` variable makes me a bit nervous - I mentally scope it to the for-loop. Could we be a bit more explicit about it and use a `last_event` variable which is updated within the loop?

Happy to merge this change as a separate PR btw.